cli/dcos-spark/security.go

```diff
@@ -0,0 +1,236 @@
+package main
+
+import (
+	"fmt"
+	"errors"
+	"crypto/rand"
+	"encoding/base64"
+	"strings"
+	"log"
+)
+
+
+const KEYLENGTH = 128
+const SECRET_REFERENCE_TEMPLATE = "spark.mesos.%s.secret.names"
+const SECRET_FILENAME_TEMPLATE = "spark.mesos.%s.secret.filenames"
+const SECRET_ENVKEY_TEMPLATE = "spark.mesos.%s.secret.envkeys"
+const sparkKdcHostnameKey = "SPARK_SECURITY_KERBEROS_KDC_HOSTNAME"
+const sparkKdcPortKey = "SPARK_SECURITY_KERBEROS_KDC_PORT"
+const sparkKerberosRealmKey = "SPARK_SECURITY_KERBEROS_REALM"
+const sparkKrb5ConfBase64BlobKey = "SPARK_MESOS_KRB5_CONF_BASE64"
+
+
+func prepareBase64Secret(secretPath string) string {
+	ss := strings.Split(secretPath, "/")
+	// The secret file without any slashes
```

Comment:
This comment is unclear. Rather rename the `fn` variable to indicate what we are extracting.